Dockerfile

```diff
@@ -135,18 +135,19 @@ RUN if [ $BACKEND = "OpenCL" ]; then \
 RUN make -j $(nproc)
 RUN make install
 
-ARG FIN_DIR=/root/dFin
-ARG FIN_TOKEN=
-#Clone Fin 
-RUN git clone https://$FIN_TOKEN:x-oauth-basic@github.com/ROCmSoftwarePlatform/Fin.git $FIN_DIR
+#Build Fin
+WORKDIR $MIOPEN_DIR
+RUN git submodule update --init --recursive
+ARG FIN_DIR=$MIOPEN_DIR/fin
 WORKDIR $FIN_DIR
 # Can be a branch or a SHA
-ARG FIN_BRANCH=4b1aecb98258252c9fb5e8e028722c9a245b98cb
-RUN git pull && git checkout $FIN_BRANCH
+ARG FIN_BRANCH=
+RUN if ! [ -z $FIN_BRANCH ]; then \
+        git pull && git checkout $FIN_BRANCH; \
+    fi
 # Install dependencies
 RUN cmake -P install_deps.cmake 
 
-#Build Fin
 WORKDIR $FIN_DIR/_hip
 RUN CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH=$MIOPEN_DEPS $FIN_DIR
```

Comment:
May check if fin exists in $FIN_DIR in the first place? (because if $FIN_BRANCH is empty, fin wouldn't be cloned, and these lines will error out)

Reply:
MIOpen now includes Fin as a submodule, so the fin directory should exist going forward. If Fin isn't included by MIOpen then this assumption fails. I think it's ok for the docker to fail when building Fin if it doesn't exist. The alternative would be to fail when checking if the folder is populated. Either way, the non-presence of Fin at this stage necessitates failure. 
This could be made more resilient by defaulting to cloning the Fin repo directly.

We would like to start using the fin included with the MIOpen commit, because it will ensure compatibility. Though I left the option to use a custom Fin commit with FIN_BRANCH.